src/Microsoft.ML.Transforms/FourierDistributionSampler.cs

```diff
@@ -105,23 +111,25 @@ public void Save(ModelSaveContext ctx)
             // int: sizeof(Float)
             // Float: gamma
 
-            ctx.Writer.Write(sizeof(Float));
+            ctx.Writer.Write(sizeof(float));
             _host.Assert(FloatUtils.IsFinite(_gamma));
             ctx.Writer.Write(_gamma);
         }
 
-        public Float Next(IRandom rand)
+        public float Next(IRandom rand)
         {
-            return (Float)Stats.SampleFromGaussian(rand) * MathUtils.Sqrt(2 * _gamma);
+            return (float)Stats.SampleFromGaussian(rand) * MathUtils.Sqrt(2 * _gamma);
         }
     }
 
     public sealed class LaplacianFourierSampler : IFourierDistributionSampler
     {
-        public class Arguments
+        public class Arguments : IFourierDistributionSamplerFactory
         {
             [Argument(ArgumentType.AtMostOnce, HelpText = "a in the term exp(-a|x| / r). r is an estimate of the average intra-example L1 distance")]
-            public Float A = 1;
+            public float A = 1;
+
+            public IFourierDistributionSampler CreateComponent(IHostEnvironment env, float avgDist) => new LaplacianFourierSampler(env, this, avgDist);
         }
```

Comment:
Is this a common pattern? Why not have Arguments not implement the IFourierDistributionSamplerFactory and instead have LaplacianFourierSampler implement this interface? Also there is no data from Arguments being used in the construction of the sampler, so its a bit odd to see this here.

Reply:
So you have two options (at least)
You can implement it in class.

```
IComponent component = new Class(new Class.Arguments()));
```
or you can do this:
```
IComponent component = new Class.Arguments();
```
Which one would you prefer?

If you worried about properties of Argument class not been used, I actually pass into Class constructor "this" object, which is argument class, and it got used in constructor

---
In reply to: [222859347](https://github.com/dotnet/machinelearning/pull/1122#discussion_r222859347) [](ancestors = 222859347)